### PR TITLE
Changes made to add the Wiley  publication family to the tool

### PR DESCRIPTION
**Pull request by lmjacoby**
1. class Wiley(JournalFamily) added 
2. get_search_query_urls ammended
   - line 159: pub_type attribute added
   - starting at line 166, parameter amended to accommodate Wiley formatting where the date range parameters are two 
     different phrases 
3. class JournalFamily: added pub_type as an attribute to handle the publication type filter option in Wiley

### CODE REVIEW

File: exsclaim/journal.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ class JournalFamily():
     open_param =        "URL parameter optionally noting open results only"
     journal_param =     "URL paremter noting journal to search"
     date_range_prarm =  "URL parameter noting range of dates to search"
+    pub_type =          "URL paramter noting publication type (specific to Wiley)"
```
Comment (trevorspreadbury) [inline exsclaim/journal.py:42]: Just make this an empty string for the base class, and describe it in a comment for now. (Or else the other journals will make their urls have "URL parameter noting..." in it

File: exsclaim/journal.py [hunk 2]
```diff
@@ -615,4 +624,81 @@ def find_captions(self, figure):
...
+
+    def get_license(self, soup):
+        open_access = soup.find('div', {'class':{'doi-access'}})
+        if open_access and 'Open Access' in open_access.text:
+            is_open = True
+            return (is_open, open_access.text)
+        return (False, "unknown")
+
+    def is_link_to_open_article(self, tag):
+        # ACS allows filtering for search. Therefore, if self.open is
```
Comment (trevorspreadbury) [inline exsclaim/journal.py:702]: ACS --> Wiley

File: exsclaim/journal.py [hunk 3]
```diff
@@ -615,4 +624,81 @@ def find_captions(self, figure):
...
+
+
+class Wiley(JournalFamily):
+    domain =            "https://onlinelibrary.wiley.com"
+    search_path =       "/action/doSearch?"
+    page_param =        "startPage="
+    max_page_size =     "pageSize=20"
+    term_param =        "AllField="
+    order_param =       "sortBy="
+    open_param =        "ConceptID=15941"
```
Comment (trevorspreadbury) [inline exsclaim/journal.py:637]: such a strange parameter

**Review by trevorspreadbury — CHANGES_REQUESTED** [review]
Looks good, just a few small changes to make

File: exsclaim/journal.py [hunk 4]
```diff
@@ -615,4 +624,81 @@ def find_captions(self, figure):
...
+        totalResults = (soup.find('span', {'class': 'result__count'}).text)
+        totalResults = int(totalResults.replace(",",""))
+
+        totalPages = math.ceil(float(totalResults/20))-1
+        page = 0
+
+    def get_additional_url_arguements(self,soup):
+        current_year = datetime.now().year
+
+        journal_link_tags = journal_list.find_all('a', href=True)
```
[the remark this answers is not in the document]
Comment (lmjacoby) [reply exsclaim/journal.py:665]: I must have somehow left it out, I don't know how it happened!

journal_list = soup_w.find(id="Published in").parent.next_sibling

File: exsclaim/journal.py (hunk 3 above)
Comment (lmjacoby) [reply exsclaim/journal.py:637]: I agree, I tried to figure out if the ConceptID=15941 was an artifact or page specific but I reproduced it every time I searched for open journals.

**Review by trevorspreadbury — APPROVED** [review]

**Pull request merged**